Create docker build for ARM based Macs

Right now the docker image is only created for x86_64 which causes problems when users run it on M1 Macs.

Running `make build` on v0.11.4 builds rust-optimizer and workspace-optimizer successfully, but on v0.11.5, I get:

```
...
docker build -t "cosmwasm/rust-optimizer":0.11.5 --file rust-optimizer.Dockerfile .
[+] Building 1.3s (3/3) FINISHED
 => [internal] load build definition from rust-optimizer.Dockerfile                                                                                                   0.0s
 => => transferring dockerfile: 961B                                                                                                                                  0.0s
 => [internal] load .dockerignore                                                                                                                                     0.0s
 => => transferring context: 2B                                                                                                                                       0.0s
 => ERROR [internal] load metadata for docker.io/cosmwasm/base-optimizer:latest                                                                                       1.1s
------
 > [internal] load metadata for docker.io/cosmwasm/base-optimizer:latest:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
make: *** [build-rust-optimizer] Error 1
```

> We added a base-optimizer image. In theory you should check out the repo and do
> 
> `make build-base-optimizer && make build-rust-optimizer`
> 
> But I am getting the same error.
> 
> @uint or @maurolacy (not sure who worked on this last)? Can you check this? Maybe it is just for M1 Macs?

I've tried this with M1. There is the error prompted.

```
[+] Building 2.7s (3/3) FINISHED
 => [internal] load build definition from rust-optimizer.Dockerfile                0.0s
 => => transferring dockerfile: 961B                                               0.0s
 => [internal] load .dockerignore                                                  0.0s
 => => transferring context: 2B                                                    0.0s
 => ERROR [internal] load metadata for docker.io/cosmwasm/base-optimizer:latest    2.6s
------
 > [internal] load metadata for docker.io/cosmwasm/base-optimizer:latest:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
make: *** [build-rust-optimizer] Error 1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## 0.12.2
+
+- Support for arm64 Docker images.
+
 ## 0.12.1
 
 - Use the Docker builder pattern to reduce image sizes by preventing temporary files from

--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,10 @@
-.PHONY: build-rust-optimizer build-workspace-optimizer build publish-rust-optimizer publish-workspace-optimizer publish
+.PHONY: build-rust-optimizer build-workspace-optimizer build create-rust-optimizer-builder use-rust-optimizer-builder publish-rust-optimizer publish-workspace-optimizer publish
 
 DOCKER_NAME_RUST_OPTIMIZER := "cosmwasm/rust-optimizer"
 DOCKER_NAME_WORKSPACE_OPTIMIZER := "cosmwasm/workspace-optimizer"
-DOCKER_TAG := 0.12.1
+DOCKER_TAG := 0.12.2
+
+# Build images locally for the host CPU architecture
 
 build-rust-optimizer:
 	docker build -t $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG) --target rust-optimizer .
@@ -10,12 +12,21 @@ build-rust-optimizer:
 build-workspace-optimizer:
 	docker build -t $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG) --target workspace-optimizer .
 
-publish-rust-optimizer: build-rust-optimizer
-	docker push $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG)
-
-publish-workspace-optimizer: build-workspace-optimizer
-	docker push $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG)
-
 build: build-rust-optimizer build-workspace-optimizer
+
+# Build multi-CPU architecture images and publish them. rust alpine images support the linux/amd64 and linux/arm64/v8 architectures.
+
+create-rust-optimizer-builder:
+	docker buildx create --name rust-optimizer-builder
+
+# create-rust-optimizer-builder must be run before running use-rust-optimizer-builder for the first time
+use-rust-optimizer-builder:
+	docker buildx use rust-optimizer-builder
+
+publish-rust-optimizer: use-rust-optimizer-builder
+	docker buildx build --platform linux/amd64,linux/arm64/v8 -t $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG) --target rust-optimizer --push .
+
+publish-workspace-optimizer: use-rust-optimizer-builder
+	docker buildx build --platform linux/amd64,linux/arm64/v8 -t $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG) --target workspace-optimizer --push .
 
 publish: publish-rust-optimizer publish-workspace-optimizer

--- a/README.md
+++ b/README.md
@@ -27,7 +27,7 @@ you to produce a smaller build that works with the cosmwasm integration tests
 docker run --rm -v "$(pwd)":/code \
   --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
   --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
-  cosmwasm/rust-optimizer:0.12.1
+  cosmwasm/rust-optimizer:0.12.2
 ```
 
 Demo this with `cosmwasm-examples` (going into eg. `erc20` subdir before running),
@@ -60,7 +60,7 @@ To compile all contracts in the workspace deterministically, you can run:
 docker run --rm -v "$(pwd)":/code \
   --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
   --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
-  cosmwasm/workspace-optimizer:0.12.1
+  cosmwasm/workspace-optimizer:0.12.2
 ```
 
 The downside is that to verify one contract in the workspace, you need to compile them
@@ -86,7 +86,7 @@ case, we can use the optimize.sh command:
 docker run --rm -v "$(pwd)":/code \
   --mount type=volume,source="devcontract_cache_burner",target=/code/contracts/burner/target \
   --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
-  cosmwasm/rust-optimizer:0.12.1 ./contracts/burner
+  cosmwasm/rust-optimizer:0.12.2 ./contracts/burner
 ```
 
 ## Development

--- a/README.md
+++ b/README.md
@@ -89,35 +89,6 @@ docker run --rm -v "$(pwd)":/code \
   cosmwasm/rust-optimizer:0.12.1 ./contracts/burner
 ```
 
-## Using on M1 Macs (ARM processor)
-
-The images on DockerHub are not yet built for ARM (see [#49]). Some people successfully use
-the x86 images through emulation. But this tends to be slow and may crash due to increased
-memory consumption.
-
-A workaround for the problem is to build the images locally on an M1 Mac.
-
-```
-$ git clone https://github.com/CosmWasm/rust-optimizer.git
-$ cd rust-optimizer
-$ git checkout v0.12.1
-$ make build
-```
-
-You now have the images available locally built for your platform.
-
-```
-$ docker image ls | grep cosmwasm/
-cosmwasm/workspace-optimizer   0.12.1        aa29e67e7612   10 seconds ago   1.03GB
-cosmwasm/rust-optimizer        0.12.1        d8d95af0a0c4   17 seconds ago   1.06GB
-```
-
-While this is better than nothing, it creates the risk that contracts cannot
-be built reproducibly. For this reason a solution to [#49] means getting
-ARM builds to DockerHub that are used by everyone.
-
-[#49]: https://github.com/CosmWasm/rust-optimizer/issues/49
-
 ## Development
 
 Take a look at the [Makefile](https://github.com/CosmWasm/rust-optimizer/blob/master/Makefile)

